lms/djangoapps/grades/tasks.py

```diff
@@ -130,9 +130,10 @@ def recalculate_course_and_subsection_grades_for_user(self, **kwargs):  # pylint
     user_id = kwargs.get('user_id')
     course_key_str = kwargs.get('course_key')
 
-    if not user_id or course_key_str:
+    if not (user_id or course_key_str):
         message = 'recalculate_course_and_subsection_grades_for_user missing "user" or "course_key" kwargs from {}'
         log.error(message.format(kwargs))
+        return
```

Comment:
I think you want to raise here, otherwise the task will be marked as succeeded.  You can omit the `log.error()` call and do a `raise Exception(message.format(kwargs))`.